accounts-management/manager.go

```diff
@@ -234,3 +236,27 @@ func (m *AccountsManager) GetVerifiedWalletAccount(address cryptotypes.Address,
 
 	return account, nil
 }
+
+func (m *AccountsManager) VerifySelectedChatAccountPassword(password string) (bool, error) {
+	m.mu.RLock()
+	defer m.mu.RUnlock()
+
+	if m.persistence == nil {
+		return false, ErrPersistenceMissing
+	}
+
+	profileKeypair, err := m.persistence.GetProfileKeypair()
+	if err != nil {
+		return false, errorspkg.Wrap(err, "error getting profile keypair")
```

Comment:
We should just return `err` in this case, the error that is coming from the db type.

```suggestion
		return false, err
```

Reply:
> the error that is coming from the db type.

That's exactly why I wrap it 😄 
I faced an error like `sql: database is closed` and it was completely unobvious where it's coming from.

Not that I am _wrapping_ it, not replacing.

Reply:
Check then `ErrKeystoreDirectoryError`, we can do something similar, like add constant `ErrCodeCannotGetProfileKeypair` then
```
func ErrGettingProfileKeypair(err error) *errors.AccountsError {
	return errors.WrapError(ErrCodeCannotGetProfileKeypair, "getting profile keypair issue", err, getErrorCategory)
}
```

and here do:
```suggestion
		return false, ErrGettingProfileKeypair(err)
```

Reply:
Completely removed the function now